tools/logger/logger.c

```diff
@@ -171,6 +174,75 @@ static int append_filter_config(struct convert_config *config, const char *input
 	return 0;
 }
 
+/* Thanks to inotify, wait 5 minutes for 'expected_file' to appear in
+ * 'watched_dir'.  Then return fopen(expected_files). fopen() failures are NOT
+ * handled; e.g. NULL from fopen() is just passed through.
+ */
+static FILE *wait_fopen(const char *watched_dir, const char *expected_file)
+{
+	if (access(watched_dir, R_OK)) {
+		fprintf(stderr, "error: %s() cannot read %s\n", __func__, watched_dir);
+		return NULL;
+	}
+
+	const int iqueue = inotify_init();
+	const int dwatch = inotify_add_watch(iqueue, watched_dir, IN_CREATE);
+	FILE *fstream;
+
+	char * const fpath = malloc(strlen(watched_dir) + 1 + strlen(expected_file) + 1);
+
+	strcpy(fpath, watched_dir);
+	strcat(fpath, "/");
+	strcat(fpath, expected_file);
+
+	/* Not racy because the inotify watch was set first. */
+	if (!access(fpath, F_OK))
+		goto fopenit;
+
+	fprintf(stdout, "%s: waiting in %s/ for %s\n", APP_NAME, watched_dir, expected_file);
+	fflush(stdout);
+
+	while (1) {
+		char evlist[1 * sizeof(struct inotify_event)];
+		const int timeout_min = 5;
+		struct pollfd pfd[1] = {{ .fd = iqueue, .events = POLLIN, .revents = 0 }};
```

Comment:
`struct pollfd pfd` would do too

Reply:
I prefer to stick to the exact API signatures and not confuse arrays with their first element. It's also more future proof.